new-log-viewer/src/utils/data.ts

```diff
@@ -1,6 +1,87 @@
 import {Nullable} from "../typings/common";
 
 
+/**
+ * Checks if 'x' is bounded by the first and last value in a sorted array of numbers.
+ *
+ * @param data Sorted array.
+ * @param x Target value.
+ * @return True if is `x` is within bounds and false if outside of bounds or array is empty.
+ */
+const isWithinBounds = (data: number[], x: number): boolean => {
```

Comment:
nit: personally I prefer `range` in the function name (i.e. `isWithinRange()`) over `bounds` as in we have methods named `decode_range()` or similar elsewhere lol. not a strong preference though

Reply:
gotcha. let's keep `bounds` then